SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestHandoverReplication/SpatialTestHandoverActorComponentReplication.cpp

```diff
@@ -0,0 +1,195 @@
+{
+	Super::PrepareTest();
+
+	AddStep(TEXT("Check initial settings"), FWorkerDefinition::AllServers, nullptr, [this]() {
+		ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(GetWorld()->GetWorldSettings());
+
+		if (AssertIsValid(WorldSettings, TEXT("World Settings of correct type")))
+		{
+			AssertTrue(IsValid(WorldSettings->GetMultiWorkerSettingsClass())
+						   && WorldSettings->GetMultiWorkerSettingsClass()->IsChildOf<USpatialTestHandoverReplicationMultiWorkerSettings>(),
+					   TEXT("MultiWorkerSettings should be of class USpatialTestHandoverReplicationMultiWorkerSettings"));
+		}
+
+		FinishStep();
+	});
+
+	AddStep(TEXT("Server 1 spawns a HandoverCube"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		HandoverCube = GetWorld()->SpawnActor<AHandoverReplicationTestCube>(HandoverReplicationTestValues::Server1Position,
+																			FRotator::ZeroRotator, FActorSpawnParameters());
+		SaveHandoverCube(HandoverCube);
+		RegisterAutoDestroyActor(HandoverCube);
+		FinishStep();
+	});
+
+	auto AddWaitingStep = [this](const FString& StepName, const FWorkerDefinition& WorkerDefinition, TFunction<void()> TickFunction) {
+		constexpr float StepTimeLimit = 10.0f;
+
+		AddStep(
+			StepName, WorkerDefinition, nullptr, nullptr,
+			[this, TickFunction](float) {
+				TickFunction();
+				FinishStep();
+			},
+			StepTimeLimit);
+	};
+
+	AddWaitingStep(TEXT("Wait until the cube is synced with all servers"), FWorkerDefinition::AllServers, [this]() {
+		RequireTrue(IsValid(HandoverCube), TEXT("Server received the cube"));
+	});
+
+	AddStep(TEXT("Modify values on the Cube to non-default values"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		HandoverCube->SetTestValues(HandoverReplicationTestValues::UpdatedTestPropertyValue);
+		HandoverCube->TestStage = EHandoverReplicationTestStage::ChangeValuesToDefaultOnGainingAuthority;
+		FinishStep();
+	});
+
+	AddWaitingStep(TEXT("Wait until updated values are received on all servers"), FWorkerDefinition::AllServers, [this]() {
+		HandoverCube->RequireTestValues(this, HandoverReplicationTestValues::UpdatedTestPropertyValue,
+										TEXT("Non-default value received on the server"));
+	});
+
+	// This step will trigger value change to the default one, after which the cube would be transported back to Server 1's authority area
+	AddStep(TEXT("Move Cube to Server 2's authority area"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		HandoverCube->SetActorLocation(HandoverReplicationTestValues::Server2Position);
+		FinishStep();
+	});
+
+	AddWaitingStep(TEXT("Wait until value is reverted to default on all servers"), FWorkerDefinition::AllServers, [this]() {
+		HandoverCube->RequireTestValues(this, GetDefault<AHandoverReplicationTestCube>()->HandoverTestProperty,
+										TEXT("Value reverted to default on the server"));
+	});
+}
+
+void ASpatialTestHandoverActorComponentReplication::SaveHandoverCube_Implementation(AHandoverReplicationTestCube* InHandoverCube)
```

Comment:
This adds moving parts in the test, that it now relies on CrossServerRPCs, and maybe the ability for an actor property to recover its value if it had no interest in the actor.
You could replace this replicated property just by a call to GetAllActorsOfClass. That would remove the dependencies on these features.

Reply:
Good thought - I'll do that.